go/vt/vttablet/tabletmanager/vreplication/controller_plan.go

```diff
@@ -50,6 +51,69 @@ const (
 	reshardingJournalQuery
 )
 
+// Check that the given WHERE clause is using at least one of the specified
+// columns with an equality or in operator to ensure that it is being
+// properly selective and not unintentionally going to potentially affect
+// multiple workflows.
+// The engine's exec function -- used by the VReplicationExec RPC -- should
+// provide guardrails for data changing statements and if the user wants get
+// around them they can e.g. use the ExecuteAsDba RPC.
+var isSelective = func(where *sqlparser.Where, columns ...*sqlparser.ColName) bool {
+	if where == nil {
+		return false
+	}
+	if len(columns) == 0 {
+		return true
+	}
+	selective := false
+	_ = sqlparser.Walk(func(node sqlparser.SQLNode) (kontinue bool, err error) {
+		switch node := node.(type) {
+		case *sqlparser.ComparisonExpr:
+			column, ok := node.Left.(*sqlparser.ColName)
+			if !ok {
+				return true, nil
+			}
+			wantedColumn := false
+			for i := range columns {
+				if columns[i].Equal(column) {
+					wantedColumn = true
+					break
+				}
+			}
+			// If we found a desired column, check that it is being used with an
+			// equality operator with = rather than != OR an in clause, logically
+			// being equal to any of N things.
+			if wantedColumn &&
+				(node.Operator == sqlparser.EqualOp && node.Operator.ToString() == sqlparser.EqualStr) ||
+				node.Operator == sqlparser.InOp {
+				selective = true  // This is a safe statement
+				return false, nil // We can stop walking
+			}
+		default:
+		}
+		return true, nil
+	}, where)
+	return selective
+}
```

Comment:
Nice one!

Reply:
Is this only limited to `controller_plan` or should this be expanded to include other places?

Reply:
I feel like it's probably a good thing to expand over time, but at this point vreplication.Engine.Exec() is the critical one and what's done here.